python/ray/tune/utils/file_transfer.py

```diff
@@ -0,0 +1,234 @@
+import io
+import os
+import shutil
+import tarfile
+
+from typing import Optional, Tuple, Dict, Generator, Union
+
+import ray
+
+
+_DEFAULT_CHUNK_SIZE_BYTES = 500 * 1024 * 1024
+
+
+def sync_dir_between_nodes(
```

Comment:
Not to be part of this PR, but would be good to have some benchmark like for this file size, takes this amount of time to sync between nodes, assuming there is not a variance (so that this benchmark is meaningful as a reference)
Also this sort of opens up the possibility of syncing arbitrarily large size files by chunking. Would also be nice to have some safety measurement to warn if slow, since everything is pretty much running on one thread.